packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,225 @@
+import { Tag } from '../cloudformation/tag';
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * Properties Tags is a dictionary of tags as strings
+ */
+export interface Tags {
```

Comment:
Does this interface translate cleanly across jsii? What will it turn into?

Reply:
(For example, it might just be an `export type = {[key: string]: string}` ?)

Reply:
I would be interested in learning where the JSII experts look to determine this. I have made the time to dig in there yet, but likely I should.

Reply:
That particular syntax element will be ignored by `jsii` I believe... so no porting to other languages... we could conceivably support it though.

Right now the best "proxy" would be to just make get and set methods... it's not very pretty in TypeScript, but... at least it'll be portable right now.

Reply:
On second thought... places it's used may actually render it as a `Map<string, whatever>`. Probably want to inspect the assembly to know for sure :)

Reply:
Just tested this out; the interface definition *as written* is interpreted by `jsii` as an empty interface (issue https://github.com/awslabs/jsii/issues/194).

Better to just write it as:

```ts
export type Tags = {[key: string]: string};
```